maraboupy/MarabouNetworkNNet.py

```diff
@@ -49,13 +54,111 @@ def __init__ (self, filename, normalize=False):
             weights          (list of list of lists) Outer index corresponds to layer
                                 number.
             biases           (list of lists) Outer index corresponds to layer number.
+
+         others:
+            inputVars
+            b_variables
+            f_variables
+            outputVars
+            numVars
+
+        Attributes from parent (MarabouNetwork)
+
+            numVars
+            equList = []
+            reluList = []
+            maxList = []
+            varsParticipatingInConstraints = set()
+            lowerBounds = dict()
+            upperBounds = dict()
+            inputVars = []
+            outputVars = np.array([])
         """
         super().__init__()
 
         self.normalize = normalize
+
+        if not filename:
+            self.numLayers = 0
+            self.layerSizes = []
+            self.inputSize = 0
+            self.outputSize = 0
+            self.maxLayersize = 0
+            self.inputMinimums = []
+            self.inputMaximums = []
+            self.inputMeans = []
+            self.inputRanges = []
+            self.outputMean = 0
+            self.outputRange = 0
+            self.weights = []
+            self.biases = []
+
+            # self.nlr = None
+            return
+
         # Read the file and load values
         self.read_nnet(filename)
 
+        self.computeNetworkAttributes()
+
+    def clearNetwork(self):
+        # call clear() from parent
+        self.clear()
+
+        self.numLayers = 0
+        self.layerSizes = 0
+        self.inputSize = 0
+        self.outputSize = 0
+        self.maxLayersize = 0
+        self.inputMinimums = []
+        self.inputMaximums = []
+        self.inputMeans = []
+        self.inputRanges = []
+        self.weights = []
+        self.biases = []
+
+    def resetNetworkFromParameters(self, inputMinimums: list, inputMaximums: list, inputMeans: list, inputRanges: list,
```

Comment:
What's the purpose of resetNetworkFromParameters? This changes some attributes, but not the equations, variables, or constraints.

Reply:
I think it does: it calls self.clear() [from MarabouNetwork] to clear these attributes, and then self.computeNetworkAttributes() to recompute the new ones.

Reply:
Ok I see that now. So if I understand correctly, this function allows users to supply their own parameters instead of reading an NNet file, which is helpful if they have a network in a different file format, for example? If so, I think we should make the input/output min/max/mean/range arguments optional. Using mean of 0 and range of 1 effectively skips normalization. Input min/max could default to float min/max, or just set to [] and edit other functions that use inputMinimimums/inputMaximums adjusted to handle an empty vector.

Reply:
Exactly, or if the network is created by splitting or slicing an already existing network represented by a MarabouNetworkNNet object. 

Making some of the arguments optional is an excellent idea! As a matter of fact, in the split-network function, some of them are currently being set to these defaults anyway (e.g. mean=0, range=1, and float min/max, precisely as you say). So it would be good to make this more systematic. I will make these changes.

Reply:
Changes made. Plus added a test that tests resetNetworkFromParameters both with default parameters and a full set of parameters, with and without normalization.